### ISSUE TITLE
BUG:  parameter encoding not work when engine='pyogrio'

### ISSUE THREAD

**Issue by CNFeffery** [geopandas/pyogrio#138]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [x] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

![image](https://user-images.githubusercontent.com/49147660/177937074-c9510e40-0f02-42aa-b290-8665613eae36.png)

#### Problem description

parameter `encoding` not work when set `engine='pyogrio'` in `gpd.read_file()`

#### Output of ``geopandas.show_versions()``

<details>

```
SYSTEM INFO
-----------
python     : 3.8.13 (default, Mar 28 2022, 06:59:08) [MSC v.1916 64 bit (AMD64)]
executable : C:\Conda\envs\geopandas-env\python.exe
machine    : Windows-10-10.0.22000-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.5.0
GDAL data dir: None
PROJ       : 9.0.1
PROJ data dir: C:\Conda\envs\geopandas-env\Library\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.11.0
pandas     : 1.4.3
fiona      : 1.8.21
numpy      : 1.23.0
shapely    : 1.8.2
rtree      : 1.0.0
pyproj     : 3.3.1
matplotlib : 3.5.2
mapclassify: 2.4.3
geopy      : None
psycopg2   : None
geoalchemy2: None
pyarrow    : None
pygeos     : 0.12.0
```

</details>

**Comment by CNFeffery** [comment]
@jorisvandenbossche  https://github.com/CNFeffery/geopandas-encoding-issues-demo/blob/main/demo.zip
![image](https://user-images.githubusercontent.com/49147660/178016037-e3f0684a-43ca-4a96-a25c-5cb39da2d374.png)

**Comment by brendan-ward** [comment]
`gdalinfo` seems to automatically detect the correct encoding of the test file.  We're first trying that ourselves via `OGR_L_TestCapability(ogr_layer, OLCStringsAsUTF8)`, but perhaps this is returning `False` in this case, in which case 
for shapefiles we are forcing to `ISO-8859-1` regardless of the `encoding` parameter.  If nothing else, we should give higher precedence to the `encoding` parameter.

**Comment by brendan-ward** [comment]
So it turns out that `OGR_L_TestCapability(ogr_layer, OLCStringsAsUTF8)` only returns `True` for shapefiles if there is a corresponding `*.cpg` file which contains `UTF-8` as its contents.

I'm not entirely sure I follow the source code for `gdalinfo` but it [looks like](https://github.com/OSGeo/gdal/blob/master/apps/ogrinfo.cpp#L413) the field names and values are simply printed directly rather than decoded to some encoding first.  

We always decode the C strings back to UTF-8 for use in Python, but here it looks like our issue is that we don't _know_ what the proper encoding is of the source file; we can't automatically detect that for shapefiles without the `*.cpg` file.  Thus the fix is to allow the caller to pass in the correct encoding, and use that as the highest precedence.  Working on a fix now...

**Comment by brendan-ward** [comment]
@CNFeffery if you need pre-built wheels with this fix before we release 0.4.1 (not yet scheduled), they are contained in the artifact.zip from our [release action](https://github.com/geopandas/pyogrio/actions/runs/2715974530).

**Comment by jorisvandenbossche** [comment]
@brendan-ward we could maybe do a 0.4.1 release? (all packaging builds on the main branch are passing, so all should be ready to tag 0.4.1, except for updating the changelog)
